src/frontend/src/optimizer/logical_optimization.rs

```diff
@@ -165,6 +165,12 @@ lazy_static! {
         ApplyOrder::TopDown,
     );
 
+    static ref JOIN_REORDER_STREAM: OptimizationStage = OptimizationStage::new(
+        "Join Reorder Stream".to_string(),
+        vec![ReorderMultiJoinRuleStreaming::create()],
+        ApplyOrder::TopDown,
+    );
```

Comment:
I suggest that we change the original join ordering rule's name to `LeftDeepTreeJoinOrderingRule` and elevate the new one to `BushyTreeJoinOrderingRule`.